src/GitVersion.Core/VersionCalculation/Abstractions/IContinuousDeliveryVersionCalculator.cs

```diff
@@ -0,0 +1,6 @@
+namespace GitVersion.VersionCalculation;
+
+public interface IContinuousDeliveryVersionCalculator
```

Comment:
Didn't we agree to reduce these interfaces down to just one, as they all have the same method signature `SematicVersion Calculate(NextVersion nextVersion)`?

Reply:
Well we discussed about that, but agreed to do the change after beta2, as it will require changing the `VersioningMode`

Reply:
Can you please explain why merging these interfaces will require changing `VersioningMode`?

Reply:
Please check this changes

https://github.com/GitTools/GitVersion/pull/3447/files#diff-d1bb8310a1a6168b70d713420c6ecc455c3bd3311b4735c3811fb8e249db8da4R61-R69

We currently have 3 `VersionMode`s but we have 4 interfaces, and @HHobeck proposed to change the `VersionMode` to 

```csharp
public enum VersioningMode
{
    ManuallyDeployment, // previously ContinuousDelivery
    ContinuousDelivery, // previously ContinuousDeployment
    ContinuousDeployment, // new
    Trunkbased // previously Mainline??
}
```

Or 

```csharp
public enum DeploymentMode
{
    ManuallyDeployment, // previously ContinuousDelivery
    ContinuousDelivery, // previously ContinuousDeployment
    ContinuousDeployment, // new
    Trunkbased // previously Mainline??
}
```

Then we have 4 modes and 4 interfaces. In that case if we define a property in a single interface like

```csharp
public interface IVersionCalculator
{
    VersioningMode VersioningMode { get; }

    SemanticVersion Calculate(NextVersion nextVersion);
}
```

each implementation of this interface will set the mode accordingly.

Reply:
I still don't understand why we can't have one interface with 4 different implementations? I would also prefer discussing renaming and other large changes in separate issues. If we are to keep a `mode` for what is basically mainline, I think we can just keep the name `Mainline`.

Reply:
Mainline is confisuing IMO because we have a `IsMainline` property as a branch configuration and a `Mainline` value on the version mode. If you say the branch needs to be mainline then it's confusing. It was just a proposal from my side because if we are touching this lines of code it's easy to do it right away. But if you don't like it nevermind.